Entities/Industry/Saw/Saw.as

```diff
@@ -182,7 +182,7 @@ bool canSaw(CBlob@ this, CBlob@ blob)
 
 		f32 dot = off * (Vec2f(0, -1).RotateBy(this.getAngleDegrees(), Vec2f()));
 
-		if (dot > 0.8f)
+		if (dot > 0.8f && (blob.getVelocity().y - this.getVelocity().y) > 0.1f)
```

Comment:
Simply checking velocity for any blob is not a reliable way to fix this and may cause too many (unforeseen) issues.

Reply:
I found that player can stand on ledge and slowly fall into a Saw without getting killed, because their velocity will be less than 0.1f. This can be fixed by changing the check to:

`blob.getVelocity().y - this.getVelocity().y) > 0.01f`

I have not found any more issues. Climbing down a ladder into the Saw will also kill the player as expected.
Are there cases where a player should get killed without moving towards the Saw?

There can be unforeseen issues with any change. That's why play-testing needs to be done.

What else can be a fix for "getting killed while changing class"?
You could make the player invincible for the first one or five ticks after changing class. Should we pursue that solution instead?

Reply:
At least checking if the blob name is knight, archer, or builder would already eliminate a lot potential issues with blobs that interact with the saw, for example bombs or trees/logs. Regardless, relying on the velocity is not how this should be approached.

Reply:
There are checks to see if the blob is tagged "flesh" or if it is an enemy mine, before the code reaches the velocity check.
Bombs/trees/logs are unaffected.